simplesamlphp-module-privacyidea/templates/LoginForm.php

```diff
@@ -155,48 +155,45 @@ class="float-l erroricon" alt="gtk-dialog-error"/>
                                 <br><br>
                                 <input id="submitButton" tabindex="1" class="rc-button rc-button-submit" type="submit"
                                        name="Submit"
                                        value="<?php echo htmlspecialchars($this->t('{login:login_button}'), ENT_QUOTES) ?>"/>
                                 <br><br>
 
-                                <!-- Hidden input which store the info about changes for future use in backend-->
+                                <!-- Undefined index is suppressed and the default is used for these values -->
                                 <input id="mode" type="hidden" name="mode"
-                                       value="<?php echo $this->data['mode'] ?: "otp" ?>"/>
+                                       value="<?php echo @$this->data['mode'] ?: "otp" ?>"/>
 
                                 <input id="pushAvailable" type="hidden" name="pushAvailable"
                                        value="<?php echo @$this->data['pushAvailable'] ?: false ?>"/>
 
                                 <input id="otpAvailable" type="hidden" name="otpAvailable"
-                                       value="<?php echo $this->data['otpAvailable'] ?: true ?>"/>
+                                       value="<?php echo @$this->data['otpAvailable'] ?: true ?>"/>
 
                                 <input id="webAuthnSignRequest" type="hidden" name="webAuthnSignRequest"
-                                       value='<?php echo $this->data['webAuthnSignRequest'] ?: "" ?>'/>
+                                       value='<?php echo @$this->data['webAuthnSignRequest'] ?: "" ?>'/>
 
                                 <input id="u2fSignRequest" type="hidden" name="u2fSignRequest"
-                                       value='<?php echo $this->data['u2fSignRequest'] ?: "" ?>'/>
+                                       value='<?php echo @$this->data['u2fSignRequest'] ?: "" ?>'/>
 
                                 <input id="modeChanged" type="hidden" name="modeChanged" value="0"/>
-                                <input id="step" type="hidden" name="step" value="<?php echo $this->data['step'] ?: 2 ?>"/>
+                                <input id="step" type="hidden" name="step"
+                                       value="<?php echo @$this->data['step'] ?: 2 ?>"/>
 
                                 <input id="webAuthnSignResponse" type="hidden" name="webAuthnSignResponse" value=""/>
                                 <input id="u2fSignResponse" type="hidden" name="u2fSignResponse" value=""/>
                                 <input id="origin" type="hidden" name="origin" value=""/>
                                 <input id="loadCounter" type="hidden" name="loadCounter"
-                                       value="<?php echo $this->data['loadCounter'] ?: 1 ?>"/>
+                                       value="<?php echo @$this->data['loadCounter'] ?: 1 ?>"/>
```

Comment:
missing HTML escaping

Reply:
is html escaping required for places where the value is not arbitrary?

Reply:
I would say yes, as values might change or attackers might get the control of the non-arbitrary values.